app/src/main/java/com/android/shelfLife/model/recipe/RecipeGeneratorOpenAIRepository.kt

```diff
@@ -32,193 +27,186 @@ import kotlinx.serialization.json.putJsonArray
-    const val LOW_CALORIE_USER_PROMPT =
-        "Create a low-calorie recipe using the following ingredients: "
+    val BASE_USER_PROMPT: (Float, String, String, Boolean, String, String) -> String =
+        { servings, recipeDescription, ingredients, onlyHousehold, recipeName, specialInstructions
+          ->
+          "Create a $recipeDescription recipe with exactly $servings servings. " +
+              "The user named the recipe \"$recipeName\" and specified the following special instructions: \"$specialInstructions\". Use the following ingredients: (${ingredients}) ${
+        if (onlyHousehold) "To ensure a valid recipe, stick strictly to the provided ingredients. Do not add any other ingredients."
+        else "To ensure a great tasting recipe, you can add other ingredients you deem relevant and necessary."
+      } Call _createRecipeFunction with a list of ingredients (name + quantity), step-by-step instructions, servings, and cooking time in seconds."
+        }
   }
 
-  // Define custom prompts for different recipe types
-  private fun getPromptsForMode(
-      listFoodItems: List<FoodItem>,
-      recipeType: RecipeType
-  ): Pair<String, String> {
+  private fun getPromptsForMode(recipePrompt: RecipePrompt): Pair<String, String> {
+    val servings = recipePrompt.servings
 
-    val foodItemsNames = listFoodItems.joinToString(", ") { it.toString() }
+    // Helper to filter and sort food items
+    val foodItems =
+        recipePrompt.ingredients.let { ingredients ->
+          if (recipePrompt.prioritiseSoonToExpire) ingredients.sortedBy { it.expiryDate }
+          else ingredients
+        }
 
-    // System and user prompts based on the recipe search type
-    return when (recipeType) {
-      RecipeType.USE_SOON_TO_EXPIRE -> {
-        USE_SOON_TO_EXPIRE_SYSTEM_PROMPT to "$USE_SOON_TO_EXPIRE_USER_PROMPT$foodItemsNames."
-      }
-      RecipeType.USE_ONLY_HOUSEHOLD_ITEMS -> {
-        USE_ONLY_HOUSEHOLD_ITEMS_SYSTEM_PROMPT to
-            "$USE_ONLY_HOUSEHOLD_ITEMS_USER_PROMPT$foodItemsNames."
-      }
-      RecipeType.HIGH_PROTEIN -> {
-        HIGH_PROTEIN_SYSTEM_PROMPT to "$HIGH_PROTEIN_USER_PROMPT$foodItemsNames."
-      }
-      RecipeType.LOW_CALORIE -> {
-        LOW_CALORIE_SYSTEM_PROMPT to "$LOW_CALORIE_USER_PROMPT$foodItemsNames."
-      }
-      RecipeType.PERSONAL -> {
-        throw IllegalArgumentException(
-            "SearchRecipeType.PERSONAL is not supported for recipe generation as it is reserved for user-created recipes.")
-      }
-    }
+    val foodItemNames = foodItems.joinToString(", ") { it.foodFacts.name }
+
+    val recipeTypeDescription =
+        when (recipePrompt.recipeType) {
+          RecipeType.BASIC -> ""
+          RecipeType.HIGH_PROTEIN -> "high protein,"
+          RecipeType.LOW_CALORIE -> "low calorie,"
+          RecipeType.PERSONAL ->
+              throw IllegalArgumentException(
```

Comment:
Is this caught somewhere? if not it might lead to bugs no?

Reply:
Good note, I put the call for getPrompts inside the try catch in generateRecipe now.